Tests/VaporOAuthTests/DefaultImplementationTests/DefaultImplementationTests.swift

```diff
@@ -9,15 +9,17 @@ class DefaultImplementationTests: XCTestCase {
         XCTAssertNil(emptyResourceServerRetriever.getServer("some username"))
     }
 
-    func testThatEmptyUserManagerReturnsNilWhenAttemptingToAuthenticate() {
-        let emptyUserManager = EmptyUserManager()
-        XCTAssertNil(emptyUserManager.authenticateUser(username: "username", password: "password"))
+    func testThatEmptyUserManagerReturnsNilWhenAttemptingToAuthenticate() async throws {
+        let emptyUserManager: UserManager = EmptyUserManager()
+        let token = try await emptyUserManager.authenticateUser(username: "username", password: "password")
+        XCTAssertNil(token)
     }
 
-    func testThatEmptyUserManagerReturnsNilWhenTryingToGetUser() {
-        let emptyUserManager = EmptyUserManager()
+    func testThatEmptyUserManagerReturnsNilWhenTryingToGetUser() async throws {
+        let emptyUserManager: UserManager = EmptyUserManager()
```

Comment:
Shouldn't `EmtpyUserManager` also be updated to be marked as `async`?

Reply:
I wasn’t sure if removing the ability for a `UserManager` to be non-async is ideal.
So by leaving `EmptyUserManager` not using `async` it indirectly tests that still works (cause compilation would fail otherwise).
Then in the test case using `: UserManager` forces the call to then be `try await` even for the `EmptyUserManager` testing the `async` case.

Let me know if you think that could be done clearer / any suggestions, thankyou!

Reply:
No let's keep it consistent and mark it as async. If the protocol defines it as async then the implementations should be async as well

Reply:
Okay, done!